zerver/lib/actions.py

```diff
@@ -317,6 +317,19 @@ class SubscriptionInfo:
 ]
 
 
+def create_historical_user_messages(*, user_id: int, message_ids: List[int]) -> None:
+    # Users can see and react to messages sent to streams they
```

Comment:
Maybe start with "Users can see and interact with messages sent to streams with public history for which they do not have a UserMessage because they were not a subscriber at the time the message was sent. In order to add emoji reactions or mutate message flags for those messages, we create UserMessage objects for those messages; these have the special `historical` flag which keeps track of the fact that the user did not receive the message at the time it was sent."

Reply:
Updated.  Feel free to tweak further before the push.